src/signal.rs

```diff
@@ -134,13 +149,47 @@ pub fn validate_signal_num(num: c_int, for_vcpu: bool) -> errno::Result<c_int> {
     Err(errno::Error::new(EINVAL))
 }
 
-/// Registers `handler` as the signal handler of signum `num`.
+/// Register the signal handler of signum.
+///
+/// # Safety
+///
+/// This is considered unsafe because the given handler will be called
+/// asynchronously, interrupting whatever the thread was doing and therefore
+/// must only do async-signal-safe operations.
 ///
-/// Uses `sigaction` to register the handler.
+/// # Arguments
 ///
-/// This is considered unsafe because the given handler will be called asynchronously, interrupting
-/// whatever the thread was doing and therefore must only do async-signal-safe operations.
-/// flags: SA_SIGINFO or SA_RESTART if wants to restart after signal received.
+/// * `num`: the signal number to be registered.
+/// * `handler`: the signal handler function to register.
+/// * `for_vcpu`: whether `num` is a vcpu signal.
+/// * `flag`: specify the behavior of the signal.
+///    Set SA_SIGINFO or SA_RESTART if wants to restart after signal received.
+///
+/// # Examples
+///
+/// ```
+/// # extern crate libc;
+/// extern crate vmm_sys_util;
+/// use libc::{c_int, c_void, siginfo_t, sigaction, SA_SIGINFO};
+/// use vmm_sys_util::signal::{register_signal_handler, SignalHandler};
+///
+/// static mut SIGNAL_HANDLER_CALLED: bool = false;
+/// extern "C" fn handle_signal(_: c_int, _: *mut siginfo_t, _: *mut c_void) {
+///     unsafe {
+///         SIGNAL_HANDLER_CALLED = true;
+///     }
+/// }
+///
+/// unsafe {
```

Comment:
I don't find this example very relevant because it registers itself for signal `0` and never gets to handle it.

Reply:
Umm, I think it would be good to move the example to block_signal because we can directly show examples for multiple APIs there.